perfdash/s3_metrics_bucket.go

```diff
@@ -17,32 +17,36 @@ limitations under the License.
 package main
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"strconv"
 	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/s3"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"k8s.io/klog"
+	"k8s.io/utils/ptr"
 )
 
 // S3MetricsBucket that creates a AWS S3 client to fetch data.
 type S3MetricsBucket struct {
-	client  *s3.S3
+	client  *s3.Client
 	bucket  *string
 	logPath string
 }
 
 // NewS3MetricsBucket creates a new S3MetricsBucket.
 func NewS3MetricsBucket(bucket, pathPrefix, region string) (MetricsBucket, error) {
+	cfg, err := awsconfig.LoadDefaultConfig(context.Background(),
+		awsconfig.WithRegion(region),
+	)
+	if err != nil {
+		return nil, err
+	}
 	return &S3MetricsBucket{
-		client: s3.New(session.Must(session.NewSession(&aws.Config{
-			Region:                        aws.String(region),
-			CredentialsChainVerboseErrors: aws.Bool(true),
```

Comment:
I couldn't find an SDK v2 replacement for `CredentialsChainVerboseErrors`, so I left it out.